gammapy/data/utils.py

```diff
@@ -0,0 +1,115 @@
+    position : `~astropy.coordinates.SkyCoord`
+        Sky position.
+    fixed_offset : `~astropy.coordinates.Angle`
+        Offset calculated from the pointing position.
+        If neither the position nor fixed_offset is specified,
+        it uses the position of the center of the map by default.
+    names : list of str
+        IRFs properties to be considered.
+        Available options are ["edisp-bias", "edisp-res", "psf-radius"]
+        (all used by default).
+    containment_fraction : float
+        Containment_fraction to compute the `psf-radius`.
+        Default is 68%.
+    standard_scaler : bool
+        Compute standardize features by removing the mean and scaling to unit variance.
+        Default is False.
+
+    Returns
+    -------
+    features : `~astropy.table.Table`
+        Features table
+
+    """
+
+    if position and fixed_offset:
+        raise ValueError(
+            "`position` and `fixed_offset` arguments are mutually exclusive"
+        )
+
+    n_obs = len(observations)
+    n_features = len(names)
+    data = np.zeros((n_obs, n_features))
+    units = [u.Unit("")] * n_features
+    for (
+        ko,
+        obs,
+    ) in enumerate(observations):
+        psf_kwargs = dict(fraction=containment_fraction, energy_true=energy_true)
+        if isinstance(obs.psf, PSFMap) and isinstance(obs.edisp, EDispKernelMap):
+            if position is None:
+                position = obs.psf.psf_map.geom.center_skydir
+            edisp_kernel = obs.edisp.get_edisp_kernel(position=position)
+            psf_kwargs["position"] = position
+        else:
+            if fixed_offset is None:
+                if position is None:
+                    offset = 0 * u.deg
+                else:
+                    offset_max = np.minimum(
+                        obs.psf.axes["offset"].center[-1],
+                        obs.edisp.axes["offset"].center[-1],
+                    )
+                    offset = np.minimum(
+                        position.separation(obs.pointing_radec), offset_max
+                    )
+            else:
+                offset = fixed_offset
+            edisp_kernel = obs.edisp.to_edisp_kernel(offset)
+            psf_kwargs["offset"] = offset
+        for kf, name in enumerate(names):
+            if name == "edisp-bias":
+                data[ko, kf] = edisp_kernel.get_bias(energy_true)
+            if name == "edisp-res":
+                data[ko, kf] = edisp_kernel.get_resolution(energy_true)
+            if name == "psf-radius":
+                containment_radius = obs.psf.containment_radius(**psf_kwargs).to("deg")
+                data[ko, kf] = containment_radius.value
+                units[kf] = u.deg
+
+    features = Table()
+    for kf, name in enumerate(names):
+        features[name] = data[:, kf]
+
+    if apply_standard_scaler:
+        features = standard_scaler(features)
+
+    features.meta = dict(
+        obs_ids=observations.ids,
```

Comment:
Could be but then it should be removed when passed to the clustering function which is inconvenient. Ideally I would prefer to use something like `Map` where obs_id and names are axes (or have a Table.rows that gives the obs_id, as Table.columns gives the names)

Reply:
I think removing the column or just ignoring it by default is fine...users also might want to drop other columns before...

Reply:
I would try to make the output of `get_irf_features` as complete as possible, even if only a subset of the information is used later. So adding a column for `obs_id` (if available) and dataset name (required) is fine. However in the most general case those could be array valued columns, because stacked observations could be grouped as well, but I think this case we can just forbid for now...